daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2258,6 +2281,48 @@ func (devices *DeviceSet) exportDeviceMetadata(hash string) (*deviceMetadata, er
 	return metadata, nil
 }
 
+// Return true only if kernel supports xfs and mkfs.xfs is available
+func supportsXFS() bool {
+        exec.Command("modprobe", "xfs").Run()
+
+        f, err := os.Open("/proc/filesystems")
+        if err != nil {
+                return false
+        }
+        defer f.Close()
+
+        s := bufio.NewScanner(f)
+        for s.Scan() {
+                if strings.Contains(s.Text(), "xfs") {
```

Comment:
`strings.HasSuffix("\txfs")` in case there are filesystems whose name include `xfs`